### PR TITLE
Add runnable tiny DNA training example

### LINKED ISSUE #412: Add a runnable, minimal DNA training example with Marin
## Goal

Give an external visitor a fast, concrete answer to: **“What does it take to train a genomic language model with MarinDNA?”**

The example should optimize for first impressions and comprehension, not for producing a useful model. A reader should be able to understand the complete data → tokenization → training path from the root README, then run the same example locally without needing private data, cloud credentials, or knowledge of our internal experiment conventions.

This should mirror the pattern used by Marin itself: a short, complete example in the README backed by a real runnable file ([upstream reference script](https://github.com/marin-community/marin/blob/fab260f8cc9a085fa2275c758d77187c908d2b30/experiments/tutorials/train_tiny_model.py#L1-L120)).

## Recommendation: README overview + runnable `examples/` directory

Do both:

1. Add a compact **“Train a tiny DNA model”** section to the root README. It should show the complete conceptual path and the exact command to run it.
2. Add the source of truth as a self-contained, executable example on `main`, for example:

   ```text
   examples/train_tiny_dna/
     README.md
     pyproject.toml
     uv.lock
     train.py
   ```

Use `examples/`, not `experiments/`. In this repository, `experiments/exp<N>_*` contains issue-specific research runs that live on permanent, unmerged branches ([repository convention](https://github.com/Open-Athena/marin-dna/blob/2d98cf559cf5c7e2e0f2bd3e72524b768c6c37d8/AGENTS.md#L37-L38)). This tutorial is durable public documentation and should remain available on `main`.

The example needs its own environment because the root installation deliberately excludes the Marin/Levanter/Iris runtime ([current README](https://github.com/Open-Athena/marin-dna/blob/2d98cf559cf5c7e2e0f2bd3e72524b768c6c37d8/README.md#L30-L48)). Keep that environment as small as the working Marin dependency graph permits.

## What the example should contain

The readable training recipe should make only the decisions needed for a tiny causal DNA language model:

- A public Hugging Face DNA dataset, deterministically capped to a small tutorial sample.
- The public MarinDNA character tokenizer.
- A tiny decoder-only model that can train on CPU.
- Standard, uniform next-token loss.
- A simple optimizer with literal hyperparameters.
- A small batch size, sequence length, and number of training steps.
- No custom evaluation suite.
- Local output under a documented `MARIN_PREFIX`.
- One copy-paste command that performs tokenization and training and produces a checkpoint.

Start with an existing public MarinDNA/Zoonomia dataset. Confirm that sampling it does not first download a large shard. If it does, publish a tiny derived tutorial dataset so the quickstart remains genuinely quick; creating that small artifact is the only dataset work in scope.

The README excerpt should be short enough to understand without opening another page. The runnable file is authoritative and may include the small amount of setup or CLI plumbing omitted from the excerpt.

## Intended user experience

From a fresh clone, the flow should be approximately:

```bash
cd examples/train_tiny_dna
uv sync
MARIN_PREFIX=local_store uv run python train.py
```

The example must run locally on CPU by default. Document the measured wall time and hardware from the validation run so users know what “tiny” means. Running on Iris, TPU, or GPU can be a short next-step link, but must not complicate the default path.

## Explicit non-goals

- Producing a biologically useful model or prescribing a production training recipe.
- Loss weighting by case, repeat status, region label, or any other feature.
- Custom VEP, likelihood-gap, or lm-eval tasks.
- Validation mixtures, dataset mixtures, hyperparameter transfer, or scaling-law logic.
- W&B as a requirement for the local run.
- Launching paid cloud resources as part of this issue.
- Turning the tutorial into an `exp<N>` research run or giving it experiment-style naming.
- General Marin or Iris installation documentation beyond what is required to execute this example.

## Validation

Validate the public path, not only individual imports:

1. Run the documented commands from a fresh environment.
2. Confirm the dataset sample is small and deterministic.
3. Confirm tokenization and at least the configured tiny training run complete on CPU.
4. Confirm a checkpoint is written under the documented local prefix.
5. Re-run and confirm Marin recognizes or reuses completed artifacts as documented.
6. Add a lightweight automated smoke test that imports the example and constructs/lowers its step graph, so Marin API drift fails visibly. Keep the full CPU run as recorded PR validation unless it is cheap and reliable enough for CI.

## Acceptance criteria

- [ ] The root README contains a concise, complete “Train a tiny DNA model” overview and links to the runnable source.
- [ ] `examples/train_tiny_dna/` is self-contained, dependency-locked, and uses the current public Marin API.
- [ ] The documented fresh-clone commands complete locally on CPU without private datasets, cloud credentials, or W&B credentials.
- [ ] The example uses a small deterministic sample of public DNA data and does not download an unexpectedly large source artifact before sampling.
- [ ] The recipe uses uniform next-token loss and contains no custom evaluations, mixtures, loss weighting, or production-only helpers.
- [ ] The run writes a checkpoint to the documented `MARIN_PREFIX`; measured runtime and validation hardware are recorded in the PR.
- [ ] A lightweight automated smoke test catches broken imports or step-graph construction after Marin dependency/API updates.
- [ ] The README and runnable example agree on dataset, tokenizer, model, and command.
- [ ] No paid compute is launched while implementing or validating this issue without explicit approval.

### PR DESCRIPTION
**Pull request by gonzalobenegas**
## Summary

- add a compact root-README example covering public enhancer data, tokenization, and tiny Qwen3 training
- add a self-contained, dependency-locked `examples/train_tiny_dna/` runnable on local CPU
- use the public `marin-dna/zoonomia-v1-v3_ccre_non_promoter-tutorial` dataset and `marin-dna/tokenizer-char-bos` tokenizer
- train a 31,360-parameter Qwen3 model for 10 steps with context length 256 and write native plus Hugging Face checkpoints below `MARIN_PREFIX`
- ignore generated caches, checkpoints, and JAX logs

## Validation

Validated from the example's locked Python 3.12.3 environment on one JAX CPU device (host: 4-vCPU Intel Xeon Platinum 8488C):

- `uv sync --locked`
- `uv lock --check`
- `MARIN_PREFIX="$PWD/local_store" WANDB_MODE=disabled uv run python train.py`
- tokenized 64 documents / 16,384 tokens in 1.78 seconds
- completed all 10 training steps; training invocation took 21.50 seconds including JIT initialization
- wrote native and Hugging Face-compatible checkpoints at step 9
- final loss: `1.5536772`; total training tokens: 10,240
- repeated the documented command; Marin pruned the completed graph and exited in 5.18 seconds
- `git diff --check`

No private data, W&B credentials, cloud resources, GPU, or TPU were used.

## Intentional scope decisions

Following review, this keeps the committed surface minimal:

- the full CPU execution above is the validation; no separate automated smoke test is added
- the root README links to the example directory, while operational commands live only in the example README

Closes #412

### CODE REVIEW

**Comment by gonzalobenegas** [comment]
🤖 Independent review completed with one actionable finding: the lockfile's pinned `jaxlib` wheel set does not support Intel macOS or musl-based Linux, while the original environment declaration could be read as supporting all macOS/Linux hosts. Resolved in [`73454d8`](https://github.com/Open-Athena/marin-dna/blob/73454d8082f29c4c40c3babd75fa5381647c6876/examples/train_tiny_dna/README.md#L12-L13) by documenting the exact supported platforms. No other actionable findings were identified.
